scripts/setup/create-database

```diff
@@ -29,8 +29,8 @@ if [ "$(su "$POSTGRES_USER" -c "cd / && psql -v ON_ERROR_STOP=1 -Atc \"SELECT 1
 fi
 
 # Shut down all services to ensure a quiescent state.
-if [ -e "/var/run/supervisor.sock" ]; then
-    su zulip -c "$(dirname "$0")/../stop-server"
+if [ -e "/home/zulip/deployments" ]; then
```

Comment:
Would it be better to look at `zulip.conf` here? I'm a bit worried about a server that used to be a Zulip server but now only has the database, but where `/home/zulip/deployments` hasn't been garbage-collected. I guess it'd be mostly harmless.

Reply:
"looking at zulip.conf" is hard, because knowing "what is a frontend host" is ... making a guess from looking at `puppet_classes`.  The only other place we try to figure this out is in `scripts/lib/install`, where we use `scripts/zulip-puppet-apply --noop --write-catalog-summary` to figure out what puppet classes would be applied.  Which seems ... very heavy-weight.  This is using the _output_ from that specific decision -- if that process detected that `zulip::app_frontend_base` is pulled in on the host, it makes a `/home/zulip/deployments`.

As you note, if this is wrong, _it's fine_.  We stop ... nothing, since if it's no a frontend host, there are no running processes to stop.

But the previous check for `supervisor.sock` is wrong, since even pure-PostgreSQL hosts have a supervisor installed.

Reply:
Yeah, I think it's reasonable to proceed as-is.